src/fun/FunctionWrapper.h

```diff
@@ -65,45 +67,47 @@ class FunctionWrapper< T(*)(Arg1, Arg2)>
     FuncType* m_fun;
 
 private:
-    struct ArgsCounter{ enum{Arg1, Arg2, COUNT}; };
+    struct ArgsCounter{ enum{I_Arg1, I_Arg2, COUNT}; };
     enum { ARGS_COUNT = ArgsCounter::COUNT};
 
 private:
     struct None;
 
-    template<int INDEX>
+    // the typename S is for avoiding error:
+    // explicit specialization in non-namespace scope
+    template<int INDEX, typename S=void>
     struct argsType{ typedef None Type; };
 
-    template<>
-    struct argsType<0>{ typedef Arg1 Type; };
+    template<typename S>
```

Comment:
adding the typename S is for avoiding error in g++: "explicit specialization in non-namespace scope".